File: topic_6a/cat.cpp

#include "cat.h"

#ifndef CAT_H
#define CAT_H

// set breed
Cat::Cat(const std::string &breed) : Pet(breed, "") {}

// set breed and name
Cat::Cat(const std::string &breed, const std::string &name) : Pet("Cat", breed, name) {}

std::string Cat::speak() const
{
    return "hiss";
}

#endif

File: java_practice/pets/Cat.java
package java_practice.pets;

public class Cat extends Pet {
    public Cat(String name, int age, int weight) {
        super(name, age, weight);
    }

    public String speak() {
        return "Meow";
    }

    public String rubHead() {
        return "Rubbing head against you.";
    }
}
